File: app/src/main/java/com/example/android/architecture/data/remote/apis/ApiProvider.java
package com.example.android.architecture.data.remote.apis;

import com.example.android.architecture.data.remote.Network;


/**
 * Created by YaoHaitao on 2018/3/12.
 */

public class ApiProvider {

    private ApiProvider() {}

    public static PostApi makePostAPI() {
        return Network.getInstance().makeApi(PostApi.class);
    }

}

File: app/src/main/java/com/example/android/architecture/utils/ApiProvider.kt
package com.example.android.architecture.utils

import com.example.android.architecture.data.remote.Network
import com.example.android.architecture.data.remote.apis.PostApi
import com.example.android.architecture.data.remote.apis.UserApi

object ApiProvider {

  fun makePostApi() = Network.makeApi(PostApi::class.java)

  fun makeUserApi() = Network.makeApi(UserApi::class.java)
}

